rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -317,6 +322,9 @@ private String findDatedSnapshotVersionIfNecessary(String groupId, String artifa
 
     @Nullable
     private MavenRepository normalizeRepository(MavenRepository originalRepository) {
+        if (!normalizeRepositories) {
```

Comment:
Made repository check (normalizeRepository) optional, because gcp artifact registry maven repositories return 404 for root repository url requests, so it was failing and the repository was getting discarded.

Reply:
I'm a bit concerned about this. Usually the reason a repository fails to normalize is that it is unreachable. And there are a _lot_ of unreachable defunct repositories. Not excluding them from the list could substantially slow down our dependency resolution.

normalizeRepository also applies credentials to a repository, so if it fails on a repo requiring auth, requests to that repo wont include the required credentials.

Is there another endpoint we could _also_ try?

Reply:
normalizeRepositories is true by default, so we continue to get the existing behavior. I just added a flag you can use to opt out when you know, like with gcp, that the normalization isn't going to work. Auth credentials are separately applied in all other places where we make requests, so skipping normalization isn't going to prevent subsequent requests from working. I have no idea if there's another endpoint that gcp artifact registry maven repos will allow; there's no documentation for http(s) endpoints.

Reply:
It looks like github package manager maven repositories do the same thing. Since maven and gradle don't make root repository url requests, it's probably not something we can expect repositories to do. You can't really treat a 404 on an artifact url as an indication that the repository is completely unusable either, since that artifact may just not be present. I'm really not sure what we can do here. Perhaps opting out when the caller knows the root url check won't work is the best option available.

Reply:
It might be nice to be able to opt out at the individual repository level, instead of opting out for all repositories in a given maven operation.

Reply:
What if there were a configurable "known to exist" list?
Servers on that list wouldn't be pinged at all by normalizeRepository. The rest of the logic in normalizeRepository could work, applying auth and such, it just wouldn't issue the request.

Reply:
I imagine our GCR repository is a unique DNS name for us, so hard to provide an accept list.

Reply:
I fixed some issues and changed the skipping of the normalization request in MavenPomDownloader to be specific to individual MavenRepository instances (knownToExist property).